### ISSUE TITLE
GenServer {Oban.Registry, {Oban, _}} terminated

### ISSUE THREAD

**Issue by jordi-chacon** [sorentwo/oban#753]
### Environment

- Oban Version: 2.13.0 (pro 0.12.2)
- PostgreSQL Version: PostgreSQL 13.6
- Elixir: 1.12.3
- Erlang/OTP: 24.0.3

```
> Oban.config()
%Oban.Config{
  dispatch_cooldown: 5,
  engine: Oban.Pro.Queue.SmartEngine,
  get_dynamic_repo: nil,
  log: :debug,
  name: Oban,
  node: "distru-worker@172.16.7.141",
  notifier: Oban.Notifiers.Postgres,
  peer: Oban.Peers.Postgres,
  plugins: [
    {Oban.Plugins.Cron,
     [
       timezone: "America/Los_Angeles",
       crontab: [...]
     ]},
    Oban.Pro.Plugins.BatchManager,
    Oban.Pro.Plugins.DynamicLifeline,
    Oban.Plugins.Reindexer,
    {Oban.Plugins.Pruner, [limit: 200000, max_age: 1200]},
    Oban.Plugins.Gossip,
    Oban.Plugins.Stager
  ],
  prefix: "public",
  queues: [
    bulk_download: [global_limit: 2],
    bulk_imports: [global_limit: 2],
    default: [global_limit: 6],
    fast: [global_limit: 6],
    high_priority: [global_limit: 6],
    ...
    scheduled: [global_limit: 2]
  ],
  repo: Distru.Repo,
  shutdown_grace_period: 15000,
  testing: :disabled
}
```

### Current Behavior

We just upgraded Oban from `2.12.1` to `2.13.0` and Oban Pro from `0.11.1` to `0.12.2`. Immediately after we started seeing `GenServer {Oban.Registry, {Oban, _}} terminated` errors and Oban jobs stopped processing. We then rolled back to our previous Oban versions and the errors stopped.

You can find the logs for all the `Oban.Registry` terminated errors here:
[GenServer Oban.Registry terminated.txt](https://github.com/sorentwo/oban/files/9235484/GenServer.Oban.Registry.terminated.txt)

It seems we got one of those errors for every queue (i.e. `{:producer, "high_priority"}`, but we also got those errors for other things like `Oban.Peer` and `{:plugin, Oban.Plugins.Stager}`.

The graph below shows job started events (`[:oban, :job, :start]`) grouped in chunks of 5 minutes:
<img width="1002" alt="Screenshot 2022-08-01 at 18 17 15" src="https://user-images.githubusercontent.com/153220/182195328-9d6507b0-dc8f-4bb9-a9fd-8679e9f5c7f4.png">

Oddly enough it looks a very small number of jobs were still being processed after the errors.

**Comment by sorentwo** [comment]
@jordi-chacon Will you please check with the recently released Pro v0.12.4? I believe fixes in that version and v0.12.3 will solve the issues you encountered.

**Comment by jordi-chacon** [comment]
@sorentwo We just deployed Pro 0.12.4 to production and it's working like a charm. Thank you so much for the fix!!
